### PR TITLE
add mkinitcpio module

### PR DESCRIPTION
**Pull request by opatut**
screenshot

### CODE REVIEW

File: src/mkinitcpio.rs [hunk 1]
```diff
@@ -0,0 +1,139 @@
+use crate::parse_args::AppConfig;
+use crate::utils::csleep;
+use rand::prelude::*;
+use rand::seq::SliceRandom;
+use yansi::Paint;
+
+const ALL_HOOKS: &'static [&str] = &[
```
Comment (svenstaro) [inline src/mkinitcpio.rs]: Do you think there's merit in putting these into a list like [here](https://github.com/svenstaro/genact/blob/master/src/main.rs#L52)? Might make it more consistent and we could also reuse these hooks for some other modules or mix them in as technical mumbo-jumbo when generating some other stuff.

File: src/mkinitcpio.rs [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+
+    msg2(
+        format!(
+            "-k /boot/vmlinuz-{preset} -c /etc/mkinitcpio.conf -g {image}",
+            preset = preset,
+            image = image
+        )
+        .as_ref(),
+    );
+    msg1("Starting build: 5.0.7-arch1-1-ARCH");
```
Comment (svenstaro) [inline src/mkinitcpio.rs]: Would be cool to vary this a bit.

File: src/mkinitcpio.rs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+    for hook in hooks {
+        msg2(format!("Running build hook: [{}]", hook).as_ref());
+        csleep(rng.gen_range(50, 1000));
+
+        if *hook == "block" && mode == "fallback" {
+            warn("Possibly missing firmware for module: aic94xx");
```
Comment (svenstaro) [inline src/mkinitcpio.rs]: Could you also vary the occurrence of these? You could also use random items from `data/cfiles.txt`. For instance, you could use up to three items from under `drivers/scsi/*.c`.

**Comment by svenstaro** [comment]
Pretty cool and simple module! Would like to merge after discussing some nitpicks I had. Also run your code through clippy. It has some helpful comments.

File: src/mkinitcpio.rs (hunk 3 above)
Comment (opatut) [reply src/mkinitcpio.rs]: Yes, sounds great!

File: src/mkinitcpio.rs (hunk 2 above)
Comment (opatut) [reply src/mkinitcpio.rs]: Found some [here](https://en.wikipedia.org/wiki/Uname). Put them in a file and randomized. :tada:

File: src/mkinitcpio.rs (hunk 1 above)
Comment (opatut) [reply src/mkinitcpio.rs]: All done.

**Comment by svenstaro** [comment]
Very, very cool! Great work.

**Pull request merged**